addons/mail/static/src/core/common/composer.js

```diff
@@ -704,6 +702,20 @@ export class Composer extends Component {
         this.suggestion?.clearRawMentions();
     }
 
+    onClickInsertCannedResponse() {
+        const composer = toRaw(this.props.composer);
+        const text = composer.text;
+        const firstPart = text.slice(0, composer.selection.start);
+        const secondPart = text.slice(composer.selection.end, text.length);
+        composer.text = firstPart + "::" + secondPart;
+        this.selection.moveCursor((firstPart + "::").length);
+        if (this.ui.isSmall && !this.env.inChatter) {
+            return false;
+        } else {
+            composer.autofocus++;
+        }
```

Comment:
Is the `false` return meant to achieve `ev.preventDefault`? If so, I’d prefer making an explicit call to `ev.preventDefault` for clarity. If not, we shouldn’t return anything and focus the composer in the `if` block.

Reply:
Not gonna lie: I just copy-pasted the `addEmoji` below but instead of `str` being the emoji this is `::`.
I can reduce code to not have the return false shenanigan, which is unused for `::` here.